### PR TITLE
[clang-cache] Make the include-tree the default and use `CLANG_CACHE_USE_CASFS_DEPSCAN` for switching to casfs depscanning

### PR DESCRIPTION
**Pull request by akyrtzi**

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: clang/test/CAS/path-independent-cas-outputs.c [hunk 1]
```diff
@@ -58,15 +58,12 @@
 // RUN: diff %t/a/t1.d %t/b/t2.d
 // RUN: diff %t/t.d %t/a/t1.d
 
-// Baseline to check we got expected output.
```
Comment (cachemeifyoucan) [inline clang/test/CAS/path-independent-cas-outputs.c:61]: why is this related to the change?

**Review by cachemeifyoucan — APPROVED** [review]
LGTM

File: clang/test/CAS/path-independent-cas-outputs.c (hunk 1 above)
Comment (akyrtzi) [reply clang/test/CAS/path-independent-cas-outputs.c:61]: The PCH emitted with include tree is not binary compatible with non-cached PCH, because it removes the header search paths that get stored in the PCH (in `HEADER_SEARCH_PATHS` record).
